src/analysis/type_enclosing.ml

```diff
@@ -23,41 +32,18 @@ let from_nodes path =
     | Module_binding_name {mb_expr = {mod_type = m}}
     | Module_declaration_name {md_type = {mty_type = m}}
     | Module_type_declaration_name {mtd_type = Some {mty_type = m}} ->
-      ret (`Modtype (env, m))
-    | _ ->
-      log ~title:"from_nodes"
-        "unhandled node under cursor: %s"
-        (Browse_raw.string_of_node node);
-      None
+      ret (Modtype (env, m))
+    | _ -> None
   in
   List.filter_map ~f:aux path
 
-let from_node env node =
-        let longident_to_string id = try
-          String.concat ~sep:"." (Longident.flatten id)
-          with Misc.Fatal_error _ -> ""
-        in
-        let ret typ = Mbrowse.node_loc node, `Type (env, typ), `No in
-        match node with
-        | Expression e ->
-          (match e.exp_desc with
-          | Texp_construct ({ Location. txt; loc=_ }, cdesc, _) ->
-            Some(longident_to_string txt, ret cdesc.cstr_res)
-          | Texp_ident (_, { Location. txt; loc=_ }, vdes) ->
-            Some(longident_to_string txt, ret vdes.val_type)
-          | _ -> None)
-        | Pattern p ->
-          (match p.pat_desc with
-          | Tpat_construct ({ Location. txt; loc=_ }, cdesc, _) ->
-            Some(longident_to_string txt, ret cdesc.cstr_res)
-          | _ -> None)
-        | _ -> None
-
-let from_reconstructed verbosity exprs env node =
+let from_reconstructed ~get_context ~verbosity env node exprs =
```

Comment:
Why is `get_context` passed as a parameter? My first impression is that it makes the code harder to reason about; `from_reconstructed` is called only once, so `get_context` is always the same why not simply inline it?